src/routes/safe/components/Transactions/TxList/TxInfoCreation.tsx

```diff
@@ -1,82 +1,70 @@
-import { Text } from '@gnosis.pm/safe-react-components'
 import { ReactElement } from 'react'
 import styled from 'styled-components'
+import { Text } from '@gnosis.pm/safe-react-components'
+import { Creation } from '@gnosis.pm/safe-react-gateway-sdk'
 
 import { getExplorerInfo } from 'src/config'
 import { formatDateTime } from 'src/utils/date'
 import { Transaction } from 'src/logic/safe/store/models/types/gateway.d'
 import { NOT_AVAILABLE } from './utils'
 import { TxDetailsContainer } from './styled'
-import { Creation } from '@gnosis.pm/safe-react-gateway-sdk'
-import { useKnownAddress } from './hooks/useKnownAddress'
+import { KnownAddressType, useKnownAddress } from './hooks/useKnownAddress'
 import PrefixedEthHashInfo from 'src/components/PrefixedEthHashInfo'
 import { TxDataRow } from './TxDataRow'
+import { md } from 'src/theme/variables'
+
+const StyledTxCreationAddress = styled.div`
+  margin-bottom: ${md};
 
-const StyledPadding = styled.div`
-  background-color: ${({ theme }) => theme.colors.white};
+  &:last-of-type {
+    margin-bottom: 0px;
+  }
 `
 
 export const TxInfoCreation = ({ transaction }: { transaction: Transaction }): ReactElement => {
   const txInfo = transaction.txInfo as Creation
   const timestamp = transaction.timestamp
 
-  const creator = useKnownAddress(txInfo.creator)
-  const factory = useKnownAddress(txInfo.factory)
-  const implementation = useKnownAddress(txInfo.implementation)
+  const generateCreatorTxData = (
+    title,
+    creationEntity: Creation['creator' | 'factory' | 'implementation'],
+    address: KnownAddressType,
```

Comment:
I assume React was complaining about conditional hooks if you moved this in here, right? Is the logic of `useKnownAddress` only a hook, or does it import a function that could be used here?

Reply:
Yes, exactly. I had to pass the `know address` to the function.
The hook is not that small so I would leave it as it is or assign the hook result to variables like it was before.
https://github.com/gnosis/safe-react/blob/25382fa939ce3dc23a273375a0d75e2400fbf7d4/src/routes/safe/components/Transactions/TxList/hooks/useKnownAddress.ts#L15-L35